added `runName` from SpaDES as a title for `scfmDiagnostics` plots

vanelslander-ecology opened the pull request:
this corresponds with an update to scfmutils that added the title option to the ggplot calls for these functions. need to pull both for them to work

vanelslander-ecology committed:
added `runName` from SpaDES as a title for `scfmDiagnostics` plots

achubaty commented:
use `.runName` parameter to pass this info from the main sim into the module for use with plotting

vanelslander-ecology committed:
Merge branch 'PredictiveEcology:development' into development

vanelslander-ecology committed:
add ifElse so plots run even if runName not supplied

vanelslander-ecology committed:
Merge branch 'PredictiveEcology:development' into development

vanelslander-ecology committed:
scfmDataPrep now saves plot of flammableMap in the outputs

vanelslander-ecology committed:
removed plot flammable Map from pr #38

vanelslander-ecology committed:
reverting previous changes

vanelslander-ecology committed:
scfmDataPrep now saves plot of flammableMap in outputs

vanelslander-ecology committed:
Merge branch 'PredictiveEcology:development' into development

vanelslander-ecology committed:
Added .runName param for plotting with titles

vanelslander-ecology commented:
wasn't entirely sure what i was doing here with the dots and .runName so let me know if i need to change anything! sorry for the trouble.

achubaty commented:
We shouldn't duplicate the code with and without the `title` argument -- the functions should handle the `title = NULL` case.

vanelslander-ecology committed:
Merge remote-tracking branch 'origin/flamPlot' into development

vanelslander-ecology committed:
Updated figure calls to include runName as title if provided

scfmutils functions can now also handle runName = NULL

vanelslander-ecology committed:
Merge branch 'PredictiveEcology:development' into development

vanelslander-ecology committed:
added .runName

vanelslander-ecology commented:
Okay I updated it so it's prepared if runName = NULL (and updated scfmutils too). Still not sure if I'm exactly right here on the .runName call so let me know.

vanelslander-ecology committed:
.runName

achubaty commented on modules/scfmDiagnostics/scfmDiagnostics.R:
```diff
@@ -39,7 +39,9 @@ defineModule(sim, list(
     defineParameter(".useCache", "logical", FALSE, NA, NA,
-                    "Should caching of events or module be used?")
+                    "Should caching of events or module be used?"),
+    defineParameter(".runName", "character", NULL, NA, NA,
```
checking `is.null()` should work below, but I think `is.na()` is clearer and it retains the character class, rather than loosely allowing the value to be character or NULL.

that said, since it's only being used to pass through as ggplot `title`, I supposed it's fine.

achubaty commented on modules/scfmDiagnostics/scfmDiagnostics.R:
```diff
@@ -39,7 +39,9 @@ defineModule(sim, list(
-                    "Should caching of events or module be used?")
+                    "Should caching of events or module be used?"),
+    defineParameter(".runName", "character", NULL, NA, NA,
+                    "Name for simulation provided by user")
```
be sure to clarify in metadata that `NULL` is allowed here, it may also be worth pointing out that `.runName` will be used for plot titles.

achubaty commented on modules/scfmDiagnostics/scfmDiagnostics.R:
```diff
@@ -84,22 +86,28 @@ doEvent.scfmDiagnostics = function(sim, eventTime, eventType) {
+      }
+      if (!exists("runName")) {
+        runName <- NULL
+      }
```
you should be able to simply assign `runName <- P(sim)$.runName` since the NULL will propagate, so you don't need these checks (NOTE: `.runName` is defined as a parameter, so use `!is.null(P(sim)$.runName)` to check if it has a non-default value)

vanelslander-ecology replied:
Okay I updated to what I think you mean. I kept the exists(runName) because it seems lots of users are providing runName as an object in their global script so I don't want to overwrite it with the .runName

achubaty replied:
it's generally bad practice to rely on variables higher up the call stack or in the global environment. it's very fragile.
users should explicitly pass this as a param.

achubaty replied:
also note that the way this is currently written, it won't use the parameter value if `runName` exists anywhere in the call stack... very prone to unexpected behaviour

achubaty commented on modules/scfmDiagnostics/scfmDiagnostics.R:
```diff
@@ -152,23 +160,35 @@ doEvent.scfmDiagnostics = function(sim, eventTime, eventType) {
+      if (!exists("runName")) {
+        runName <- NULL
+      }
+
```
simply use `runName <- P(sim)$.runName` per above

vanelslander-ecology committed:
Updated runName calls

vanelslander-ecology committed:
runName as aparam

vanelslander-ecology committed:
Merge branch 'PredictiveEcology:development' into development

vanelslander-ecology committed:
Updated .runName Param to account for NULL issue

vanelslander-ecology committed:
added runName to burn maps
